src/dialog/llm/__init__.py

```diff
@@ -20,5 +26,40 @@ def get_llm_class() -> Type[AbstractLLM]:
         llm_class = getattr(llm_module, class_name)
     except Exception as e:
         logging.info(f"Failed to load LLM class {Settings().LLM_CLASS}. Using default LLM. Exception: {e}")
+        raise
+
+    if isinstance(llm_class, AbstractLLM):
+        return llm_class, "AbstractLLM"
+
+    elif 'langchain_core.runnables' in str(type(llm_class)):
+        return llm_class, "LCELRunnable"
+
+    logging.info(f"Type for LLM class is: {type(llm_class)}")
+
+    return DialogLLM, "AbstractLLM"
+
+
+def process_user_message(message, chat_id=None):
+    LLM, llm_type = get_llm_class()
+    if llm_type == "AbstractLLM":
+        llm_instance = LLM(config=Settings().PROJECT_CONFIG, session_id=chat_id)
+        ai_message = llm_instance.process(message.message)
+
+    elif llm_type == "LCELRunnable":
```

Comment:
could it be easier to maintain the previous DialogLcelLLM and make its interface gradually equal to a langchain runnable, avoiding this kind of handler? This seems to create more complexity to find a common ground sometime.

As an example of implementation, for LCEL wrapper, configurables and other configuration parameters would be initialized in the init, process method could call a chain's invoke method, and the chain could be fully constructed with runnables For example
```
class DialogLcelLLM(AbstractLLM):
    def __init__(self, ...):
        super().__init__(...)
        # handle init parameters in each class case
     def generate_prompt(...):
     # creates a ChatPromptTemplate 
     def llm(...)
     # the llm component
     def chain(...)
     # this is new, could represent better the LCEL chain using the correct name
     def process(...)
     # this takes the input question and processes, in LCEL case it woudl call self.chain.invoke(inputs)
     def postprocess(...)
     # custom postprocessing
```

Reply:
Well, it would have been easier in the past, before this whole implementation hahahaha. If we do agree to this common ground, I can rollback without any problems.

IMHO, I think we need to have support for both structures, allowing users to easily migrate to our solution, but that's what I think.

Reply:
@lgabs what do you think: let's let the user use this structure as it is and then we implement the class in a future PR (probably this week, getting the DialogLcelLLM class back), or do you think this is a no go in this PR?